### PR TITLE
Add compressOnly mode and also update README to discuss forks

### LINKED ISSUE #53: Better documentation and/or handling of forks
First up, thanks for the excellent GitHub action!

However, I was surprised recently to find this doesn't work on pull requests from forks and fails silently, with only this error message hidden in the GitHub action logs to indicate it didn't complete:

```
    - Processing: /github/workspace/src/static/images/test.png
->> Generating markdown…
->> Committing files…
	 *  Head SHA: 41a31c3fc57433f287e5091d538e79504bb4310c
	 *  Tree 5849f57c3558964e83b965ed66b32332a8e077ac
	 *  Converting images to blobs…
(node:1) UnhandledPromiseRejectionWarning: HttpError: Resource not accessible by integration
    at response.text.then.message (/usr/local/src/image-actions/node_modules/@octokit/request/lib/request.js:56:27)
    at processTicksAndRejections (internal/process/task_queues.js:86:5)
(node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:1) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Digging into GitHub actions I understand why – because upstream repos don't have access to change downstream branches – and on second look, I did find this mentioned on the README:

> The GITHUB_TOKEN secret is automatically generated by GitHub. This automatic token is scoped only to the repository that is currently running the action.

However this was not clear to me, and imagine it isn't to others either and accepting pull requests from forks is really what GitHub is all about.

I've a few suggestions to improve this:

**Making people more aware of this restriction**

First of all I would suggest expanding on that line in the README to explain what it means to forks. I'm happy to throw in a suggested pull request with wording for this if you'd like?

We could also add advice to only run this GitHub action on pull requests from and to the same repo, as pointless run other than that so a waste of time and compute. Not sure exactly how to do this but happy to look into the appropriate `if:` line to add to the suggested .yml config if you want?

**How can we fix this?**

I presume getting forked repos to implement a [Personal Access Token](https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token) would work, but not sure that's a great idea to be honest. It only works for regular contributors who will go to this effort of setting up a PAT – and regular contributors might as well work off of the main repo.

A better option might be  to allow the action to also run on pushes. Then could run it on master after the PR has been approved. Ideally that would allow the option to commit directly to the push branch (master in this case - for those that are crazy enough to want that) or to open a new pull request with the compressed images after merging (probably better!).

Alternatively changing the action to only do the compression, and not handle the commit, would allow the [Checkout GitHub Action](https://github.com/actions/checkout) to handle the commit (though this does an extra config line to checkout the correct head branch), or something like the [Create Pull Request GitHub Action](https://github.com/marketplace/actions/create-pull-request) to create a new pull request, after merging pull requests from forks. Might mean less code for you to maintain, more flexibility, but at the cost of more complicated workflow set up. Or alternatively make this optional for those (like me!) that want the power of handling the commit differently. Probably need to do that to maintain backwards compatibility to be honest.

Anyway, quite a lot in there, so maybe better split into a number of issues but thought best to start a discussion first so let me know your thoughts.

### PR DESCRIPTION
**Pull request by bazzadp**
Fixes #53 

I'm not sure how to test this to be honest as couldn't get the compression running either locally or as a GitHub action. I tested as best I could the bits without the compression bits and the code I added seemed to work, but could do with a good looking over. Or a test docker image being built if possible (sorry I'm not familiar with Docker at all!)?

Most of the changes however are to the README and the number of code changes are actually quite small.

I also moved the config load from `image-processing.js` to `index.js` as needed one bit of config in there and didn't want to load it twice.

### CODE REVIEW

**Comment by bazzadp** [comment]
Oops not sure why I closed this one! Reopening...

**Comment by bazzadp** [comment]
Gentle nudge @benschwarz !

**Comment by benschwarz** [comment]
@bazzadp, apologies for the huge delay here.

In review, I'm happy with the changes made on this, but I'd like to discuss an alternate option that I feel will improve the overall usage of image-actions. 

Instead of `compressOnly`, I was thinking we could split out two configuration options:  

- `postMarkdownSummary` (default to `true`)
- `commitChanges` (default to `true`)

What do you think? 

-- 

Closes #50

File: src/index.js [hunk 1]
```diff
@@ -21,6 +24,16 @@ const run = async () => {
   console.log("->> Generating markdown…");
   const markdown = await generateMarkdownReport(results);
 
+  // Expose the markdown to an Action output
+  const escaped_markdown = markdown.replace(/\%/g,'%25').replace(/\n/g,'%0A').replace(/\r/g,'%0D')
```
Comment (benschwarz) [inline src/index.js:28]: Is there a documentation reference that can be referenced with this comment?

Comment (bazzadp) [reply src/index.js:29]: Added!

**Comment by bazzadp** [comment]
> Instead of `compressOnly`, I was thinking we could split out two configuration options:
> 
> * `postMarkdownSummary` (default to `true`)
> * `commitChanges` (default to `true`)
> 
> What do you think?

Hmmm... I'm not completely sure.

Why would you want to post a markdown summary to a pull_request (the only postMarkDownSummary that works) if you aren't also committing the changes to that pull_request? Doesn't the messaging imply the images have been compress? And notnjsut compress and (effectively) uncompressed again.

And on the flip side I'm not sure why you'd want to commit changes, but not post the summary?

We always put the markdown into `outputs.markdown` if you want to use it separately (e.g. I'm proposing not committing for forked pull requests and instead running on master and then opening a new PR for that with the summary, for my use case for example). so the summary is always available to you whether you commit or not, so people can do extra things if they like with it, if that was your concern?

**Comment by bazzadp** [comment]
Any more thoughts here @benschwarz ?

**Comment by bazzadp** [comment]
I saw [this comment](https://github.com/calibreapp/image-actions/issues/60#issuecomment-678652129):
>An option to comment-only or keep the PR from merging if an image is found larger than max width would be nice.

Is that where the desire for comment-only version comes from @benschwarz ?

That kind of makes sense, but my concern would be that you could fill up the PR with comments if you continue to push other changes to that branch that don't fix the images and run the action multiple times. Could become very messy very quickly. Updating a comment would be an alternative solution, but that would be more difficult and may not be wanted if the comment was not the latest comment. Or update if latest, and add if not latest? But then no notification for the update. All in all it seems a good bit of extra complexity so requires some extra thought so not keen to make part of this PR now.

Another alternative is to have a "fail" option on the Action and let people look at the logs for the detail if it does fail - so have the action act like a check that only optimised images are committed. That would probably be considerably less complicated, and could possibly be added to this branch if you want?

Alternatively we merge this as is and iterate.

Let me know your thoughts.

File: src/index.js [hunk 2]
```diff
@@ -21,6 +24,17 @@ const run = async () => {
   console.log("->> Generating markdown…");
   const markdown = await generateMarkdownReport(results);
 
+  // Expose the markdown to an Action output
+  // https://github.community/t/set-output-truncates-multiline-strings/16852
+  const escaped_markdown = markdown.replace(/\%/g,'%25').replace(/\n/g,'%0A').replace(/\r/g,'%0D')
```
Comment (benschwarz) [inline src/index.js:29]: Could you update `escaped_markdown` to be camelCase?

File: action.yml [hunk 3]
```diff
@@ -32,6 +32,15 @@ inputs:
     required: false
     default: "node_modules/**"
 
+  compressOnly:
+    description: "Only runs the compression, without adding to Pull Request"
```
Comment (benschwarz) [inline action.yml:36]:
```suggestion
    description: "Images will be compressed. No commit, or comments will be added to your Pull Request"
```

File: README.md [hunk 4]
```diff
@@ -1,6 +1,12 @@
 # Image Actions
 
+<<<<<<< HEAD
+Image Actions automatically compress JPEG, PNG and WebP images in GitHub Pull Requests within the same repository.
+
+It can also run in "Compress Only" mode to allow this to be used for non-Pull Requests, and also for Pull Requests from forks.
+=======
 Image Actions automatically compresses JPEG, PNG and WebP images in GitHub Pull Requests.
+>>>>>>> upstream/master
```
Comment (benschwarz) [inline README.md:9]: There's enough documentation lower in the readme, could you remove this from the introduction section?

**Comment by benschwarz** [comment]
> > An option to comment-only or keep the PR from merging if an image is found larger than max width would be nice.
> 
> Is that where the desire for comment-only version comes from @benschwarz ?

Yep, but let's get this PR merged and tackle other ideas later. 
I've posted a couple of small nitpicks, but otherwise I'm happy for this to be merged ASAP.

File: README.md (hunk 4 above)
Comment (bazzadp) [reply README.md]: Done.

File: action.yml (hunk 3 above)
Comment (bazzadp) [reply action.yml]: Done

File: src/index.js (hunk 2 above)
Comment (bazzadp) [reply src/index.js]: Done!

**Comment by bazzadp** [comment]
Updated so think we're good to merge.

**Comment by benschwarz** [comment]
Thanks @bazzadp !

**Pull request merged**